src/main/java/com/puppycrawl/tools/checkstyle/PackageObjectFactory.java

```diff
@@ -164,35 +189,105 @@ private Object createObjectFromMap(String name) throws CheckstyleException {
     }
 
     /**
-     * Create a new instance of a named class.
-     * @param className the name of the class to instantiate.
-     * @param secondAttempt the set of names to attempt instantiation
-     *                      if usage of the className was not successful.
-     * @return the {@code Object} created by loader or null.
-     * @throws CheckstyleException if the class fails to instantiate.
+     * Generate the map of third party Checkstyle module names to their fully qualified names.
+     * @param loader the class loader used to load Checkstyle package names
+     * @return the map of third party Checkstyle module names to their fully qualified names
      */
-    private Object createObjectWithIgnoringProblems(String className, Set<String> secondAttempt)
-            throws CheckstyleException {
-        Object instance = createObject(className);
-        if (instance == null) {
-            final Iterator<String> ite = secondAttempt.iterator();
-            while (instance == null && ite.hasNext()) {
-                instance = createObject(ite.next());
-            }
+    private Map<String, String> generateThirdPartyNameToFullModuleName(ClassLoader loader) {
+        Map<String, String> returnValue;
+        try {
+            final ClassPath classPath = ClassPath.from(loader);
+            // in packages, the value is something like "com.puppycrawl.tools.checkstyle."
+            // the last "." need to be dropped
```

Comment:
@Luolc Is this an issue with [PackageNamesLoader.getPackageName](https://github.com/checkstyle/checkstyle/blob/b4a337a599e853cdc44b19a76da28fcfab2956cd/src/main/java/com/puppycrawl/tools/checkstyle/PackageNamesLoader.java#L105)? Should we change it there to drop or not add the last period?

@romani Is there any reason for the last period? class is not api.